src/basic/strverscmp.c

```diff
@@ -1,20 +1,75 @@
 /* SPDX-License-Identifier: LGPL-2.1-or-later */
 
+#include "strverscmp.h"
+
+#if !defined SD_BOOT
 #include <ctype.h>
 
 #include "string-util.h"
-#include "strverscmp.h"
 
-static bool is_alpha(char a) {
+typedef bool sd_bool;
+#define sd_strcmp(a, b)     strcmp((a), (b))
+#define sd_strncmp(a, b, n) strncmp((a), (b), (n))
```

Comment:
hmm, can't we instead just define strcmp to StrCmp() on efi? i.e. use the regular C names for these instead of prefixing them?

I'd probably try to do that for most stuff, i.e. consider the uefi naming the anomaly and always map the uefi names back to the regular C stuff, where that is reasonable